Improve gamepads

BenoitRoux0 opened the pull request:
Improve gamepads game speed by using a second thread for updates loop.

BenoitRoux0 committed:
Add gamepad support and create separate thread for input handling

Added functionalities to handle gamepad inputs and run it on a separate thread for better responsiveness. The thread updates gamepads and their inputs while the game is running. Also, pthread library has been incorporated into the Makefile to support thread creation and handling. To ensure smooth handling of gamepad inputs, mutexes have been initialized and clear gamepad function has also been updated to clean up allocated mutexes effectively. Also, the 'update_inputs_bonus.c' file has been revised to properly register the inputs that have been read. The list of gamepad sources in sources.mk has been updated to include 'read_inputs_bonus.c'. The hooks_bonus.c is revised to call read_inputs instead of update_inputs. Furthermore, player input via gamepad has been adjusted to include the functionality of pressing LB or RT to trigger weapon animation while in gamepad mode.

gd-harco reviewed (changes requested):
Currently have a data race (see #65)

BenoitRoux0 committed:
Remove hard-coded compiler and improve thread safety

Removed the explicit gcc compiler definition in the Makefile to ensure flexibility and portability. Added mutex locks on the open function in gamepad update to prevent potential race conditions and improve application stability.

gd-harco reviewed (commented):
data race seemed fixes
